### PR TITLE
Add a new relation with etcd charm

### PR DESCRIPTION
**Pull request by capsali**

### CODE REVIEW

File: lib/Modules/ComputeHooks/ComputeHooks.psm1 [hunk 1]
```diff
@@ -659,6 +669,41 @@ function Get-HGSContext {
     return $ctxt
 }
 
+function Get-EtcdContext {
+    Write-JujuWarning "Generating context for etcd"
+    $required = @{
+        "client_ca" = $null
```
Comment (gabriel-samfira) [inline lib/Modules/ComputeHooks/ComputeHooks.psm1:675]: Is SSL always enabled in etcd? Is there a situation where you may only get non SSL info on this relation?

If SSL is optional on this relation, you need to move the certificate stuff in the ```-Optional``` option, and make sure you account for the changes bellow.

File: lib/Modules/ComputeHooks/ComputeHooks.psm1 [hunk 2]
```diff
@@ -659,6 +669,41 @@ function Get-HGSContext {
...
+    # Write etcd certs
+    $etcd_ca_file = Join-Path $NOVA_INSTALL_DIR "etc\etcd-ca.crt"
+    $etcd_cert_file = Join-Path $NOVA_INSTALL_DIR "etc\etcd-client.crt"
+    $etcd_key_file = Join-Path $NOVA_INSTALL_DIR "etc\etcd-client.key"
+    # Remove the current certificates (if any) and add the new ones
+    Remove-Item -Recurse -Force "$etcd_ca_file" -ErrorAction SilentlyContinue
+    Remove-Item -Recurse -Force "$etcd_cert_file" -ErrorAction SilentlyContinue
+    Remove-Item -Recurse -Force "$etcd_key_file" -ErrorAction SilentlyContinue
+    # Write the new certificates
+    $ctx["client_ca"] | Set-Content $etcd_ca_file
```
Comment (gabriel-samfira) [inline lib/Modules/ComputeHooks/ComputeHooks.psm1]: ```Set-Content $etcd_ca_file $ctx["client_ca"]```

File: templates/stein/nova.conf [hunk 3]
```diff
@@ -113,7 +113,7 @@ html5_proxy_base_url = {{ html5_proxy_base_url }}
 [oslo_messaging_notifications]
 driver = messagingv2
 
-{% if compute_cluster_driver -%}
+{% if compute_cluster_driver and backend_url -%}
 [coordination]
-backend_url = etcd3+http://{{ etcd_host }}:{{ etcd_port }}
+backend_url = {{ backend_url }}
```
Comment (gabriel-samfira) [inline templates/stein/nova.conf:118]: <3

File: lib/Modules/ComputeHooks/ComputeHooks.psm1 (hunk 1 above)
Comment (capsali) [reply lib/Modules/ComputeHooks/ComputeHooks.psm1:675]: AFAIK ssl is always enabled on the etcd charm. There is no way of telling it to not be SSL from the config

**Pull request merged**